### PR TITLE
TDengine support

### LINKED ISSUE #20070: support TDengine as a stand-alone database in driver list
**Is your feature request related to a problem? Please describe.**

TDengine is an open-source and cloud-native time series database. It has gained 21k+ stars on GitHub and 255k+ installations from over 50 countries. TDengine supports SQL standards and provides a generic JDBC driver to access. DBeaver can be configured with a generic driver to access TDengine and works very well.
But the user needs to configure the driver strings manually now. I would like to ask the DBeaver team to add TDengine as a stand-alone database with default configuration strings to make the user experience better and simpler.

**Describe the solution you'd like**

To list TDengine in a predefined time-series database list with pre-configured connection strings. Note TDengine supports both native and REST connections. The native connection uses a 6030 port and requires the user to install the client library. The REST connection uses a 6041 port and does not need the user to install the library.

_I would suggest DBeaver supports both of them. One can be TDengine and another one can be TDengine-REST._

REVISED: I would suggest DBeaver supports REST interface only to reduce the burden user needs to install the client library. 

**Describe alternatives you've considered**

Still, use a generic JDBC driver but not good for the user.

**Additional context**

TDengine can be installed by the command 'docker run -d -p 6030:6030 -p 6041:6041 tdengine/tdengine:latest'. 
I am the contact person of the TDengine team. I will do my best to assist if needed.

Thanks,

### PR DESCRIPTION
**Pull request by sangshuduo**
TDengine is an open-source and cloud-native time series database. It has gained 21k+ stars on GitHub and 255k+ installations from over 50 countries. TDengine supports SQL standards and provides a generic JDBC driver to access. DBeaver can be configured with a generic driver to access TDengine and works very well.
But the user needs to configure the driver strings manually now. I would like to ask the DBeaver team to add TDengine as a stand-alone database with default configuration strings to make the user experience better and simpler.

To enable accessing TDengine in DBeaver, I added two drivers in this pull request.

# 1. TDengine driver
One is to access a TDengine instance installed on-premises via the RESTful interface.

You can use a test instance to verify. Please fill the Host with '49.232.86.127', Username with 'root', and Password with 'taosdata'.
image

Then you can run a simple SQL command to verify.
image

# 2. TDengine Cloud driver
Another one is to access the TDengine Cloud service via the URL and authorization by the token that TDengine Cloud assigned.

You can use a cloud test instance to verify. Please fill the URL with "jdbc:TAOS-RS://gw.us-east.azure.cloud.tdengine.com?useSSL=true&token=e22feb6c35f847b73fd94543af8b12f0f8f6f37b". Please note the instance might take about 20 seconds to wake up if it has been idle.
image

And run SQL command to verify as below.
image

Thanks,
Shuduo from TDengine team
mailto:sdsang@taosdata.com

### CODE REVIEW

**Comment by sangshuduo** [comment]
https://github.com/dbeaver/dbeaver/issues/20070

**Comment by LonwoLonwo** [comment]
Hello @sangshuduo 

So, the cloud version does not require any authentication at all? Even user name and password?

**Comment by sangshuduo** [comment]
> Hello @sangshuduo
> 
> So, the cloud version does not require any authentication at all? Even user name and password?

The cloud instance will provide a unique token after a user signed up. The token is for authentication. That's why no need username and password.

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml [hunk 1]
```diff
@@ -954,6 +954,38 @@
...
+                    class="com.taosdata.jdbc.rs.RestfulDriver"
+                    sampleURL="jdbc:TAOS-RS://{host}:{port}/[{db}]"
+                    defaultPort="6041"
+                    description="TDengine driver"
+                    supportedConfigurationTypes="MANUAL"
+                    webURL="http://www.tdengine.com"
+                    icon="icons/tdengine_icon.png"
+                    iconBig="icons/tdengine_icon_big.png"
+                    categories="timeseries">
+                    <file type="jar" path="https://repo1.maven.org/maven2/com/taosdata/jdbc/taos-jdbcdriver/3.2.1/taos-jdbcdriver-3.2.1-dist.jar"/>
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]: Please do not use direct maven repo URLs.
Use artifact references like
`<file type="jar" path="maven:/taosdata:taos-jdbcdriver:RELEASE[XXX]"/>`
similarly to all other artifact references in generic plugin.

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml (hunk 1 above)
Comment (sangshuduo) [reply plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]: fixed

**Review by serge-rider — APPROVED** [review]

**Comment by sangshuduo** [comment]
@serge-rider Hi, may I know when this PR will be merged and which release would include the support? Not push at all. I just want to delete my test instance since they are public access now. Thanks.

**Comment by serge-rider** [comment]
@sangshuduo most likely it will be included in the next version of Community Edition (23.1.1). Thank you for the contribution 👍

**Comment by Matvey16** [comment]
@sangshuduo Hello, is this intended to not be able to select a database in SQL Editor? I assume it is entended as object names always need to be fully qualified in queries. Correct?

https://github.com/dbeaver/dbeaver/assets/82543000/f934d2ee-64af-4bd7-88ac-234d419ec510

**Comment by sangshuduo** [comment]
> @sangshuduo Hello, is this intended to not be able to select a database in SQL Editor? I assume it is entended as object names always need to be fully qualified in queries. Correct?
> 
> dbeaver\_W6vVGq9oYN.mp4

@Matvey16 Hello, I referred another generic driver and modified some parameters for tdengine. I don't know what was missed or changed as I made a mistake if the database can be selected. Could you please advise? Thanks.

**Pull request merged**